pkgs/by-name/sd/sdl3/package.nix

```diff
@@ -0,0 +1,221 @@
+{
+  lib,
+  stdenv,
+  config,
+  alsa-lib,
+  apple-sdk_11,
+  cmake,
+  darwinMinVersionHook,
+  dbus,
+  fcitx5,
+  fetchFromGitHub,
+  ibus,
+  installShellFiles,
+  libGL,
+  libdecor,
+  libdrm,
+  libjack2,
+  libpulseaudio,
+  libusb1,
+  libxkbcommon,
+  mesa,
+  nas,
+  ninja,
+  nix-update-script,
+  nixosTests,
+  pipewire,
+  sndio,
+  systemdLibs,
+  testers,
+  validatePkgConfig,
+  wayland,
+  wayland-scanner,
+  xorg,
+  zenity,
+  alsaSupport ? stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isAndroid,
+  dbusSupport ? stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isAndroid,
+  drmSupport ? stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isAndroid,
+  ibusSupport ? stdenv.hostPlatform.isUnix && !stdenv.hostPlatform.isDarwin,
+  jackSupport ? stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isAndroid,
```

Comment:
Does it actually cross-compile to this entire zoo of targets?

Reply:
Doesn't cross-compile in general, though the issues don't seem to be in this derivation at least.

Android:

```
error: Package ‘glu-9.0.3’ in /home/puna/.cache/nixpkgs-review/pr-326699/nixpkgs/pkgs/by-name/me/mesa_glu/package.nix:56 is marked as broken, refusing to evaluate.
```

`libGLU` -> `gst-plugins-bad` -> `gtk4` -> `ibus` -> `sdl3`

---

Regular aarch64-linux:

```
CMake Error at CMakeLists.txt:4 (find_package):
  Could not find a package configuration file provided by "ECM" (requested
  version 0.0.11) with any of the following names:
```

`xcb-imdkit` -> `fcitx5` -> `sdl3`

Reply:
> Does it actually cross-compile to this entire zoo of targets?

In theory it could -- but as described above, not currently

This was just adapted from the current [SDL2](https://github.com/NixOS/nixpkgs/blob/d237fee665ceffa4f8a7dda1286396762a1659dc/pkgs/development/libraries/SDL2/default.nix#L11-L43) packaging, though

Reply:
Then please remove it. It's a huge amount of complexity and basically none of this actually needs to be conditional.

Reply:
> It's a huge amount of complexity

Is it? I don't think specifying "linux but not android" as opposed to just "linux" is much more complicated -- likewise "unix but not darwin". If this was an actually complicated expression for the sake of cross compilation I'd understand, but these are just a logical AND. I'm perfectly fine with maintaining this

> It's a huge amount of complexity and basically none of this actually needs to be conditional.

This is completely incorrect. Even if we were to only support tier 1 targets, dbus, drm, ibus, and jack are all exclusive to Linux and would be conditional. Now if you mean they don't need to be arguments/override options....I guess? But I don't see much gain from inlining these into the expression when we can instead give people options on what they want to build SDL with

Further, keeping these options actually helps a lot in solving build/eval (for native or cross) issues if our defaults (currently) fail. Using the issues @OPNA2608 came across above as an example:

> Android:
>
> ```
> error: Package ‘glu-9.0.3’ in /home/puna/.cache/nixpkgs-review/pr-326699/nixpkgs/pkgs/by-name/me/mesa_glu/package.nix:56 is marked as broken, refusing to evaluate.
> ```
> libGLU -> gst-plugins-bad -> gtk4 -> ibus -> sdl3
> 

This can be fixed with just `.override { ibusSupport = false; }`

```console
> pkgsCross.aarch64-android.sdl3.override { ibusSupport = false; }
«derivation /nix/store/5cgngvsw5i9944d7a83sq26ff4cl1a18-sdl3-aarch64-unknown-linux-android-3.1.6.drv»
```

(note: I didn't actually build this, so it or other packages could still fail, but that's not really my point here)

> Regular aarch64-linux:
>
> ```
> CMake Error at CMakeLists.txt:4 (find_package):
>   Could not find a package configuration file provided by "ECM" (requested
>   version 0.0.11) with any of the following names:
> ```
> xcb-imdkit -> fcitx5 -> sdl3

This can also be fixed by disabling ibusSupport as well. Even got it building:

```console
> nix-build --expr 'with import ./. { config = { }; overlays = [ ]; }; pkgsCross.aarch64-multiplatform.sdl3.override { ibusSupport = false; }'
/nix/store/b6jnsb7ijs88n54ly56lf34xmdifn0vs-sdl3-aarch64-unknown-linux-gnu-3.1.6-lib
```

These options also help in creating smaller, more special builds; not everyone needs libudev, dbus, xorg, libdecor, jack, etc. support.

I also disagree with the idea we should be removing cross compilation support for packages because their dependencies are currently broken. That simply makes the problem worse, and pushes more work off to the future if those dependencies were to ever be fixed